source/slang/slang-ast-print.cpp

```diff
@@ -75,7 +80,27 @@ void ASTPrinter::_addDeclPathRec(const DeclRef<Decl>& declRef)
     if (auto aggTypeDeclRef = parentDeclRef.as<AggTypeDecl>())
     {
         _addDeclPathRec(aggTypeDeclRef);
-        sb << ".";
+        sb << toSlice(".");
+    }
+    else if (auto namespaceDeclRef = parentDeclRef.as<NamespaceDecl>())
+    {
+        _addDeclPathRec(namespaceDeclRef);
+        // Hmm, it could be argued that we follow the . as seen in AggType as is followed in some other languages
+        // like Java.
+        // That it is useful to have a distinction between something that is a member/method and something that is
+        // in a scope (such as a namespace), and is something that has returned to later languages probably for that
+        // reason (Slang accepts . or ::). So for now this is follows the :: convention.
+        //
+        // It could be argued them that the previous '.' use should vary depending on that distinction.
```

Comment:
FWIW I personally think of `.` as the correct operator in Slang, and `::` is just there for compatibility. Sometimes I will use `::` when discussing language features to draw the distinction between `someObj.someInstanceMethod` and `SomeType::someInstanceMethod` where the latter is a static reference to a per-instance method, but I'm not aware of any cases where a `::` operator is strictly *needed* in the language.

The main beneficial thing `::` provides in C++ is an ability to explicitly name the global scope (Rust ran into the problem at some point of having no way to unambiguously name a top-level module - not sure how they fixed it). The big problem is that using `::` as an infix operator at times and a prefix operator-like thing at other times is itself a source of ambiguity.

If possible I'd like to avoid sending our users down the C++-minded `::` path any more than they are already inclined to go that way. However, if we discover a strong reason why having `::` as a distinct scoping operator is important then I will gladly retract all of this.